src/Solver.cpp

```diff
@@ -489,52 +655,212 @@ void Solver::initializeCmfd(){
+  _timer->recordSplit("Total time");
+}
+
 
-    if (FSR_segment_tallies[r] == 0) {
-      log_printf(ERROR, "No tracks were tallied inside FSR id = %d. Please "
-                 "reduce your track spacing, increase the number of "
-                 "azimuthal angles, or increase the size of the FSRs", r);
+/**
+ * @brief Computes the total source distribution by performing a series of 
+ *        transport sweep and source updates.
+ * @details This is the main method exposed to the user through the Python
+ *          interface to compute the source distribution, e.g., for a fixed
+ *          and/or external source calculation. This routine makes an initial 
+ *          guess for the scalar and boundary fluxes and performs transport 
+ *          sweeps and source updates until convergence. 
+ *
+ *          By default, this method will perform a maximum of 1000 transport
+ *          sweeps with a 1E-5 threshold on the integrated FSR total source. 
+ *          These values may be freely modified by the user at runtime.
+ *
+ *          The k_eff parameter may be used for fixed source calculations
+ *          with fissionable material (e.g., start-up in a reactor from
+ *          a fixed external source). In this case, the user must "guess"
+ *          the critical eigenvalue to be be used to scale the fission source.
+ *
+ *          The res_type parameter may be used to control the convergence
+ *          criterion - SCALAR_FLUX, TOTAL_SOURCE (default) and FISSION_SOURCE
+ *          are all supported options in OpenMOC at this time.
+ *
+ *          This method may be called by the user from Python as follows:
+ *
+ * @code
+ *          // Assign fixed sources
+ *          // ...
+ * 
+ *          // Find the flux distribution resulting from the fixed sources
+ *          solver.computeFlux(max_iters=100, k_eff=0.981)
+ * @endcode
+ *
+ * @param max_iters the maximum number of source iterations to allow
+ * @param k_eff the sub/super-critical eigenvalue (default 1.0)
+ * @param res_type the type of residual used for the convergence criterion
+ */
+void Solver::computeSource(int max_iters, double k_eff, residualType res_type) {
+
+  if (_track_generator == NULL)
+    log_printf(ERROR, "The Solver is unable to compute the source "
+               "since it does not contain a TrackGenerator");
+
+  else if (k_eff <= 0.)
+    log_printf(ERROR, "The Solver is unable to compute the source with "
+               "keff = %f since it is not a positive value", k_eff);
+
+  log_printf(NORMAL, "Computing the source...");
+
+  /* Clear all timing data from a previous simulation run */
+  clearTimerSplits();
+
+  /* Start the timer to record the total time to converge the flux */
+  _timer->startTimer();
+
+  _k_eff = k_eff;
+  FP_PRECISION residual;
+
+  /* Initialize data structures */
+  initializePolarQuadrature();
+  initializeExpEvaluator();
+  initializeFluxArrays();
+  initializeSourceArrays();
+  initializeFSRs();
+
```

Comment:
You should add `countFissionableFSRs();` in case the user wants to use `FISSION_SOURCE` for convergence.